src/main/webapp/user_profile_script.js

```diff
@@ -12,9 +12,145 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+async function getInfo(keyString) {
+    const queryURL = "/tasks/info?key=" + keyString;
+    const request = new Request(queryURL, {method: "GET"});
+    const response = await fetch(request);
+    const info = await response.json();
+    console.log(info[0]);
+    return info[0];
+}
+
+async function deleteTask(keyString) {
+    const info = await getInfo(keyString);
+    if (info.status !== "OPEN") {
+        window.alert("You can only delete an 'OPEN' task.")
+    } else {
+        const queryURL = "/tasks?key=" + keyString;
+        const request = new Request(queryURL, {method: "DELETE"});
+        const response = await fetch(request);
+    }
+}
+
+async function editTask(keyString) {
+    const info = await getInfo(keyString);
+    console.log(info);
+    if (info.status !== "OPEN") {
+        window.alert("You can only edit an 'OPEN' task.")
+    } else {
+        document.getElementById("edit-content-input").value = info.detail;
+        document.getElementById("edit-point-input").value = info.reward.toString();
+        const id_input = document.getElementById("task-id-input");
+        id_input.value = info.keyString;
+        id_input.readOnly = true;
+        id_input.style.display = "none"
+        document.getElementById("editTaskModal").style.display = "block";
+    }
+}
+
 function deleteRow(row) {
     if (confirm("Are you sure that you want to delete this already published task?")) {
         var rowIndex = row.parentNode.parentNode.rowIndex;
         document.getElementById("need-help").deleteRow(rowIndex);
     }
+}
+
+function deleteInProgress() {
```

Comment:
I don't think this function or the editInProgress function ever get used/called, since the same alert is used within the editTask and deleteTask functions

Reply:
So `editInProgress` and `deleteInProgress` are two functions that I prepare for the fake version of my user profile page. Now that I have implemented the real functionality, I will remove them in the next commit.

Reply:
ahhh that makes sense, sounds good!